src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,214 @@
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isInSwitch(ast) && isWhiteSpaceBeforeColon(ast)) {
+            log(ast, MSG_KEY, ast.getText());
+        }
+    }
+
+    /**
+     * Checks if the colon is inside a switch block.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if colon case is inside a switch block.
+     */
+    private static boolean isInSwitch(DetailAST colonAst) {
+        return TokenUtil.isOfType(colonAst.getParent(), TokenTypes.LITERAL_CASE,
+                TokenTypes.LITERAL_DEFAULT);
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case or switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhiteSpaceBeforeColon(DetailAST colonAst) {
+        final DetailAST parent = colonAst.getParent();
+        final boolean isColonOfCase = parent.getType() == TokenTypes.LITERAL_CASE;
+        final boolean isColonOfDefault = parent.getType() == TokenTypes.LITERAL_DEFAULT;
+        return isColonOfCase && isWhitespaceBeforeColonOfCase(colonAst)
+                || isColonOfDefault && isWhitespaceBeforeColonOfDefault(colonAst)
+                || isColonOnDifferentLine(colonAst);
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhitespaceBeforeColonOfCase(DetailAST colonAst) {
+        final DetailAST previousSibling = colonAst.getPreviousSibling();
+        int offset = 0;
+        if (previousSibling.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {
+            offset = 1;
+        }
+        return colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + offset;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhitespaceBeforeColonOfDefault(DetailAST colonAst) {
+        final boolean horizontalWhitespace;
+        final DetailAST commentBlock = colonAst.getPreviousSibling();
+        if (commentBlock == null) {
+            final DetailAST literalDefault = colonAst.getParent();
+            horizontalWhitespace = colonAst.getColumnNo()
+                    != literalDefault.getColumnNo() + literalDefault.getText().length();
+        }
+        else {
+            horizontalWhitespace =
+                    colonAst.getColumnNo() != getLastColumnNumberOf(commentBlock) + 1;
+        }
+        return horizontalWhitespace;
```

Comment:
make it `return result;`
name of variable is too confusing.

Reply:
Done.
Yes, it was confusion. I didn't change it since it was not pointed in earlier PR.

https://github.com/wilcoln/checkstyle/blob/c5fd71c27ccdb7485c08853517019e60484a15db/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java#L135